crates/noirc_evaluator/src/ssa/conditional.rs

```diff
@@ -567,3 +710,27 @@ fn create_if_subgraph(ctx: &mut SsaContext, prev_block: BlockId) -> (BlockId, Bl
 
     (new_entry, new_exit)
 }
+
+#[derive(Debug)]
+struct Segment {
+    a: (usize, NodeId),
+    b: (usize, NodeId),
```

Comment:
These could be named better. It's unclear what `a` and `b` are for here. It is also unclear what `usize` represents. Right now it is some opaque number next to a NodeId.

Reply:
I renamed as left/right. The usize is the node position in their list